src/main/java/core/basesyntax/model/io/DataConverterImpl.java

```
@@ -0,0 +1,27 @@
+package core.basesyntax.model.io;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> lines) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        for (String line : lines) {
+            String[] parts = line.split(",");
+
+            if (parts.length != 3) {
+                continue;
```

Silently continuing when `parts.length != 3` masks input problems. The task requires validating input data and handling incorrect data (e.g., quantity less than zero or invalid operation). Instead of quietly skipping, throw an informative RuntimeException or log and fail fast so tests can detect malformed lines. Quote from checklist: "Incorrect data in the input file, for example, quantity less than zero or incorrect strategy".